Mifare.c

```diff
@@ -1,203 +1,406 @@
-        return PyErr_Format(SelectError, "ApplyProtocolSettings command failed: %04x", status);
-    }
 
-    /* Empty the pAtqa */
-    memset(pAtqa, '\0', 2);
+PyObject *Mifare_init(Mifare *self, PyObject *args, PyObject *kwds) {
+	
+	int ret;
+	ret = Set_Interface_Link();
+	if(ret != 0)
+	{
+		return PyErr_Format(ReadError, "Hardware Initialisaton failed: %04x", ret);
+	}
+	Reset_reader_device();
+
+	ret = NfcRdLibInit();
+		if(ret != 0)
+	{
+		return PyErr_Format(ReadError, "Hardware Initialisaton failed: %04x", ret);
+	}
+    Py_RETURN_NONE;
+}
 
-    if ((status = phpalI14443p3a_RequestA(&self->data.I14443p3a, pAtqa)) != PH_ERR_SUCCESS) {
-        return PyErr_Format(SelectError, "RequestA command failed: %04x", status);
-    }
 
-    /* Reset the RF field */
-    if ((status = phhalHw_FieldReset(&self->data.hal)) != PH_ERR_SUCCESS) {
-        return PyErr_Format(SelectError, "FieldReset command failed: %04x", status);
-    }
-    /* Activate the communication layer part 3 of the ISO 14443A standard. */
-    if ( phpalI14443p3a_ActivateCard(&self->data.I14443p3a, NULL, 0x00, byteBuffer, &byteBufferSize, &bSak, &bMoreCardsAvailable) == PH_ERR_SUCCESS) {
-        // Card is present, return it as a python object.
-        uint8_t i;
-        char asciiBuffer[UID_ASCII_BUFFER_SIZE];
-
-        if (byteBufferSize + 1 > UID_ASCII_BUFFER_SIZE) {
-            // Truncate if we got back too much data
-            byteBufferSize = UID_ASCII_BUFFER_SIZE - 1;
-        }
+PyObject *Mifare_select(Mifare *self)
+{
+phStatus_t  status = 0;
+    uint16_t    wTagsDetected = 0;
+    uint8_t     bUid[PHAC_DISCLOOP_I3P3A_MAX_UID_LENGTH];
+    uint8_t     bUidSize;
+    while(1)    /* Continuous loop */
+    {
 
-        for (i = 0; i < byteBufferSize; i++) {
-            sprintf(&asciiBuffer[2 * i], "%02X", byteBuffer[i]);
+        
+        do
+        {
+            /* Field OFF */
+            status = phhalHw_FieldOff(pHal);
+            CHECK_STATUS(status);
+
+            /* Configure Discovery loop for Poll Mode */
+            status = phacDiscLoop_SetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_NEXT_POLL_STATE, PHAC_DISCLOOP_POLL_STATE_DETECTION);
+            CHECK_STATUS(status);
+
+            /* Run Discovery loop */
+            status = phacDiscLoop_Run(&sDiscLoop, PHAC_DISCLOOP_ENTRY_POINT_POLL);
+
+        }while((status & PH_ERR_MASK) != PHAC_DISCLOOP_DEVICE_ACTIVATED); /* Exit on Card detection */
+
+        /* Card detected */
+        /* Get the tag types detected info */
+        status = phacDiscLoop_GetConfig(&sDiscLoop, PHAC_DISCLOOP_CONFIG_TECH_DETECTED, &wTagsDetected);
+
+        /* Check for Status */
+        if ((status & PH_ERR_MASK) == PH_ERR_SUCCESS)
+        {
+            /* Check for Type A tag detection */
+            if (PHAC_DISCLOOP_CHECK_ANDMASK(wTagsDetected, PHAC_DISCLOOP_POS_BIT_MASK_A))
+            {
+
+							uint8_t byteBufferSize = sDiscLoop.sTypeATargetInfo.aTypeA_I3P3[0].bUidSize + 1;
```

Comment:
Spacing appears to be way off in this block.  Please use spaces, NOT tabs, and a 4-space indentation level.